File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {

      unordered_map< int , int > count;

    // int count[nums.size()];

      for(int i = 0 ; i < nums.size(); i++ )
      {
        if(count[nums[i]] == 0  )
        {
            count[nums[i]] = 1;
        }
        else{
            return true;
        }
      }

      return false;  

    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.java
class Solution {
    public int longestConsecutive(int[] nums) {
        
        if(nums.length == 0 )
        {
            return 0;
        }

        int count = 1 ;
        int maxi = 1;
        Arrays.sort(nums);
        
        for(int i = 0 ; i < nums.length -1 ;i++)
        {
            if(nums[i+1]== nums[i] + 1){
                count++;
            }
            else if(nums[i + 1] == nums[i])
            {
                continue;
            }
            else
            {
                maxi = Math.max(maxi , count);
            
                //reset count to 1 
                count = 1;
            }
        }
        return Math.max(maxi , count);
    }
}
